SonarQube Refactoring Sprint 7

rdsaliba opened the pull request.

rdsaliba committed:
Refactoring part 1

Fixing minor and major code smells, code convention. Added logger to the console.

rdsaliba committed:
Update pom.xml

rdsaliba committed:
refactoring form input validation to reduce logical complexity error in SonarQube

rdsaliba committed:
Reducing code smells, making the code compliant

rdsaliba committed:
Resolving generic exceptions, code smells and reducing another logical complexity

rdsaliba committed:
Reduced HotSpot Security issue + Logging everywhere

rdsaliba committed:
Delete 0eb26ea9dbbb2bc082d743ba852f979966fa5391

rdsaliba committed:
Delete 1adf75d879e33c59b8a2368c8c5487030e966e03

rdsaliba committed:
Delete 4493a558e7652ec0fcae58ed4cbb1525cf53032c

rdsaliba committed:
Delete 2d86b4767c763607a1a045ca6a90892fd5ee78bd

rdsaliba committed:
Delete 54701aceb5b612fc07206efd5f78bf768fd6c652

rdsaliba committed:
Update ModelStrategy.java

Removed Logger, was added by mistake

rdsaliba committed:
Resolving last Security HotSpot caused by SQL injection not safe

rdsaliba committed:
Added logger inside needed classes

Removing the logger from the interface class caused the class implementing it to not work. So the logger was added within the classes that needed it. Cause by previous commit, SonarQube doesn't like to have a Logger constant within a interface.

JayChouteau commented on JavaFXUserInterface/src/main/java/external/AssetDAOImpl.java:
```diff
@@ -19,12 +21,14 @@
 
 public class AssetDAOImpl extends DAO implements AssetDAO {
 
+    Logger logger = LoggerFactory.getLogger(AssetDAOImpl.class);
```
SonarQube doesn't like this logger because the parent class (DAO) has a variable with the same name (logger). There should be a name differentiation for the logger between the parent class (DAO) and all the child classes (DAOImpl)

JayChouteau replied:
Actually, it would be preferable to just remove any declaration of a logger in the child classes.

rdsaliba replied:
Hi Jeremie. This is actually an omission from my side. The correct way is to only declare the variable in the parent class to allow any other class that extends it to use the logger declaration. This will be part of my next commit.

JayChouteau commented on Server/pom.xml:
```diff
@@ -14,23 +14,11 @@
-            <url>file:///${project.basedir}\local-maven-repo</url>
-        </repository>
-    </repositories>
-
```
It seems like my project cannot resolve the dependency without this repository. Can we keep it? It doesn't seem to be causing an issue on SonarQube?

rdsaliba replied:
Where you able to resolve the issue on your local project? I was able to successfully fix the dependencies on my IDE without the need for the local-maven-repo. Please reach out if you need help.

JayChouteau replied:
I think the problem with removing the local maven repo is that the wek Deepl 1.7.0 dependency isn't resolvable online. This is why we need a local maven repo in the first place. However, since you installed that repo on your computer, this dependency is resolved. However, I don't think this would work on someone who clones the repo for the first time. @Khal-Shah could you have a look at this please?

Khal-Shah replied:
As mentioned by @JayChouteau, the local maven repo is needed to hold the wekaDeeplearning4j-1.7.0 jar which is utilized to build the LSTM model. This jar file is a plugin for the weka library that isn't available in the central maven repository.

In our test folder, you can try to run the LSTMTest (UnitTests > rul > models) and you'll most likely get errors because a lot of dependencies will be missing. In contrast to the develop branch, where it will pass. Please reinstate it.

JayChouteau replied:
I added the local repository back. Maybe Khal and other people could give a look at this as well.

JayChouteau commented on Server/src/main/java/local/AssetDAOImpl.java:
```diff
@@ -19,6 +21,8 @@
 
 public class AssetDAOImpl extends DAO implements AssetDAO {
 
+    Logger logger = LoggerFactory.getLogger(AssetDAOImpl.class);
```
Remove this declaration as the logger is already declared in the parent class

JayChouteau commented on Server/src/main/java/local/AssetTypeDAOImpl.java:
```diff
@@ -6,12 +6,18 @@
 
 public class AssetTypeDAOImpl extends DAO implements AssetTypeDAO {
+
+    Logger logger = LoggerFactory.getLogger(AssetTypeDAOImpl.class);
```
Remove this declaration as it is already declared in the parent class (DAO)

JayChouteau commented on Server/src/main/java/local/ModelDAOImpl.java:
```diff
@@ -19,6 +21,9 @@
 
 public class ModelDAOImpl extends DAO implements ModelDAO {
+
+    Logger logger = LoggerFactory.getLogger(ModelDAOImpl.class);
```
Remove this declaration as it is already declared in the parent class (DAO)

JayChouteau commented on Server/src/main/java/local/RulDAOImpl.java:
```diff
@@ -7,11 +7,15 @@
 import java.sql.SQLException;
 
 public class RulDAOImpl extends DAO implements RulDAO {
+    Logger logger = LoggerFactory.getLogger(RulDAOImpl.class);
```
Remove this declaration as it is already declared in the parent class (DAO)

JayChouteau reviewed (changes requested):
I ran the SonarQube Analysis and the work was well done. The logger addition seems to be working properly. Also, most of the code smells left are duplicated code due to our division of DAOs. I left some comments for the remaining code smells. Security hotspots have been dealt with, nice job. With all these changes, the application still functions as intended. The naming changes to variables and packages are good. Other review comments:
1) Logger: it seems that all logger were setup to give an error message. I didn't go through all of them, but is it normal. Should some of them have logger.warning or logger.info or else instead. It is possible that they are all meant for errors as well.

JayChouteau commented on JavaFXUserInterface/src/main/java/external/AssetDAOImpl.java:
```diff
@@ -57,7 +61,7 @@ public void deleteAssetByID(int assetID) {
             }
         } catch (SQLException e) {
-            e.printStackTrace();
+            logger.error("Exception in getAllLiveAssets(): ", e);
```
To target the duplication issues: This method can be removed (both from the implementation java file and interface file) as it is never used in the UI module (aside from a test that can be deleted as well).

JayChouteau replied:
The same comment for the createAssetFromQueryResult() method.

rdsaliba replied:
I'm not sure to understand your suggestion here. Please reach out to me for this, I can include the change in the next commit.

JayChouteau replied:
I pushed a commit to resolve this

JayChouteau reviewed (changes requested):
I added a few comments for code dupplication

rdsaliba committed:
Removing useless logger from child classes and cleaning dependencies

rdsaliba committed:
Merge branch 'develop' into SonarQube_Refactoring_Sprint7

rdsaliba committed:
Fixing merge conflicts

JayChouteau committed:
Removing unused code + quick touch-ups (#227)

JayChouteau committed:
Code Cleanup - UI Module (#227)

JayChouteau committed:
Code Cleanup - Server Module (#227)

JayChouteau committed:
Merging new develop (#227)

JayChouteau committed:
Resolving added SonarQube issues (#227)
- Adding back the local maven repo
- Fixing SonarQube issues added in last pr merge

JayChouteau committed:
Resolving loggers + 1 security hotspot (#227)

JayChouteau committed:
1 logger security fixed (#227)

jeffbuonamici commented on JavaFXUserInterface/src/main/java/controllers/AssetsController.java:
```diff
@@ -306,7 +308,7 @@ public void generateList() {
     /**
      * Stops the timeline
      */
-    private void closeTimeline() {
+    public void closeTimeline() {
         if (rulTimeline != null)
             rulTimeline.stop();
     }
```
This method is no longer used, and can probably be deleted.

JayChouteau replied:
It was deleted

jeffbuonamici commented on JavaFXUserInterface/src/test/java/e2e/com/cbms/app/AssetInfoTest.java:
```diff
@@ -63,12 +63,12 @@ public void start (Stage stage) throws Exception {
-        AssetDAOImpl assetDAOImpl = new AssetDAOImpl();
-        ArrayList<Asset> assets = assetDAOImpl.getAllLiveAssets();
-        AssetInfoController.initData(assets.get(0));
-        stage.setTitle("CBMS");
-        stage.setScene(scene);
-        stage.show();
+//        AssetDAOImpl assetDAOImpl = new AssetDAOImpl();
+//        ArrayList<Asset> assets = assetDAOImpl.getAllLiveAssets();
+//        AssetInfoController.initData(assets.get(0));
+//        stage.setTitle("CBMS");
+//        stage.setScene(scene);
+//        stage.show();
```
Do we need this anymore?

rdsaliba replied:
Mmh, I will remove this.

Khal-Shah commented:
I am currently getting the following error if I try running the server: java: package org.slf4j does not exist

jeffbuonamici reviewed (commented):
Really good work. I like how you broken down the input validation.

rdsaliba commented:
Some changes introduced another issue...  I will fix this 
<img width="1332" alt="image" src="https://user-images.githubusercontent.com/17152773/111007199-4a32b880-835c-11eb-9e41-121e2cc465d4.png">

rdsaliba commented:
> I am currently getting the following error if I try running the server: java: package org.slf4j does not exist

Wanna go on discord, we can take a look. @Khal-Shah

jeffbuonamici commented:
I get the following error when running the RealTimeDemo
![image](https://user-images.githubusercontent.com/11137499/111007846-c5489e80-835d-11eb-8404-f5d76b2804da.png)

rdsaliba commented:
> I get the following error when running the RealTimeDemo

This has to do with the same error I just reported. Should be easily fixed by making the Logger public from the DAO. @jeffbuonamici

rdsaliba committed:
removing dead code

jeffbuonamici commented:
I also get this error when running the server.
![image](https://user-images.githubusercontent.com/11137499/111008946-4b65e480-8360-11eb-8cd7-705e90335a3e.png)

rdsaliba committed:
Reverting the code to develop version

The following was causing an issue on the raw data tab. Was not populating the cells.

rdsaliba commented:
The issue where the raw data tab was not being populated is now fixed.

jeffbuonamici commented:
There's a merge conflict that needs to be fixed.

rdsaliba committed:
Merge branch 'develop' into SonarQube_Refactoring_Sprint7

Khal-Shah reviewed (changes requested):
The local-maven-repo folder is still not there. Please check the "Files Changed" tab to see if it has been added remotely and not just on your end. Also run the unit test "LSTMTest" to check whether you have the right dependencies and a working local-maven-repo.

I am also currently getting this error when I try to run the main in server:
<img width="373" alt="image" src="https://user-images.githubusercontent.com/35305423/111030885-af28f580-83d2-11eb-9aa6-275282ab06a2.png">

TalalBazerbachi reviewed (changes requested):
I am also getting the same error as Khalid, I can't test to see if the previous errors have been solved

rdsaliba committed:
Placing the SplashScreenPreloader inside utilities class

rdsaliba committed:
adding back local-maven-repo dependency

rdsaliba committed:
addind back weka jar

rdsaliba committed:
Create wekaDeeplearning4j-1.7.0.jar

rdsaliba committed:
Update pom.xml

rdsaliba committed:
Resolving more sonarqube issues generated by recent merges

rdsaliba committed:
Removing useless method

JayChouteau commented:
@Khal-Shah @t98b @jeffbuonamici All your comments and concerns should have been addressed with the last commits that Roy pushed. Could you confirm? On our side:
- LSTMTEST could be run
- Server could be run
- Ui could be run
- Real-Time Demo could be run
- SonarQube issues introduced with Najim PR were fixed
- The local maven repo and the pom file is back to how it was 

@jeffbuonamici when you run the server, make sure you have the latest dump file executed (V15)

PaulMicu commented:
I dont understand how this PR has 34 commits, is it possible to make a new PR and group the commits into one so we can read the code better

rdsaliba committed:
Refactoring all issues from SonarQube

rdsaliba committed:
Merge branch 'SonarQube_Refactoring_Sprint7' of https://github.com/rdsaliba/notorious-eng into SonarQube_Refactoring_Sprint7

TalalBazerbachi reviewed (commented):
are we not supposed to change xml files?
